framework/src/outputs/XDA.C

```diff
@@ -44,15 +44,17 @@ XDA::XDA(const InputParameters & parameters) :
 void
 XDA::output(const ExecFlagType & /*type*/)
 {
+  std::string fname = filename();
+  std::string base = fname.substr(0, fname.size()-4);
   if (_binary)
   {
-    _mesh_ptr->getMesh().write(filename()+"_mesh.xdr");
-    _es_ptr->write (filename()+".xdr", ENCODE, EquationSystems::WRITE_DATA | EquationSystems::WRITE_ADDITIONAL_DATA);
+    _mesh_ptr->getMesh().write(base + "_mesh.xdr");
```

Comment:
I'm just saying that you have duplicated logic in both methods
```
if (_binary)
    // Append ".xdr";
  else
    // Append ".xda";
```

I can merge as is, it just interesting that you throw the suffix away and then have it hardcoded to put it back on again.